src/scripts/main.js

```diff
@@ -1,5 +1,32 @@
 'use strict';
+console.log(document);
+const spider = document.querySelector('.spider');
```

Comment:
Ensure that the element with the class 'spider' exists in the document. If it doesn't, `spider` will be `null`, and accessing properties like `offsetParent` will cause a runtime error.